### ISSUE TITLE
Suspected regression in kubevirt v0.53.0-rc.0 (live migration on upgrade)

### ISSUE THREAD

**Issue by orenc1** [kubevirt/kubevirt#7664]
**What happened**:
When trying to consume the newly released kubevirt v0.53.0-rc.0 in HCO, all upgrade tests were failing because the virt-launchers were not upgraded on kubevirt upgrade, despite having `LiveMigrate` on `.spec.workloadUpdateStrategy.workloadUpdateMethods` at KubeVirt CR.
As a result, post-upgrade there are VMIs with `kubevirt.io/outdatedLauncherImage` label that are still running.
The VM that is being used in the test is defined here:
https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/hack/vm.yaml
which is live-migratable.

**What you expected to happen**:
All live-migratable VMs` virt-launchers should be upgraded, and not VMIs with `kubevirt.io/outdatedLauncherImage` label should be present on the cluster post-upgrade.

**How to reproduce it (as minimally and precisely as possible)**:
Constantly failing on all upgrade lanes over multiple cloud providers in the CI of:
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/1918

**Additional context**:

**Environment**:
- KubeVirt version (use `virtctl version`): v0.53.0-rc.0
- Kubernetes version (use `kubectl version`) :v1.23.3-2047+d464c70a480f1c-dirty
- VM or VMI specifications: [link](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/hack/vm.yaml)
- Cloud provider or hardware configuration: AWS, Azure and GCP
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

**Comment by nunnatsa** [comment]
@davidvossel

**Comment by orenc1** [comment]
Note:
Full KubeVirt CR post upgrade is available here:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kubevirt_hyperconverged-cluster-operator/1918/pull-ci-kubevirt-hyperconverged-cluster-operator-main-hco-e2e-upgrade-index-azure/1521232245312983040#1:build-log.txt%3A11105

in particular, the `workloadUpdateStrategy` configuration is present:
```
         "workloadUpdateStrategy": {
            "batchEvictionInterval": "1m0s",
            "batchEvictionSize": 10,
            "workloadUpdateMethods": [
                "LiveMigrate"
            ]
        } 
```

**Comment by tiraboschi** [comment]
Maybe some kind of "conflict" with the new MigrationPolicy mechanism?

**Comment by davidvossel** [comment]
We have a presubmit [e2e test](https://github.com/kubevirt/kubevirt/blob/a1220326fcb6b16d03feeae74c48f21b0e4d9a74/tests/operator_test.go#L1442 ) that validate this exact scenario works updating from the latest release to the latest code in main. It hasn't been failing. We're only guaranteeing updates from n-1 releases.

I tested this locally using your VM spec just to validate we aren't missing anything, and the vmi updated as expected when updating from v0.52.0 to v0.53.0.

What version of KubeVirt are you all updating from?

**Comment by nunnatsa** [comment]
@davidvossel - maybe this is the issue: we couldn't make the v0.52.0 bump to pass the smoke test and it was never merged, so HCO is still using v0.51.0

**Comment by orenc1** [comment]
We are testing upgrade from same HCO minor version (1.7.0), which currently contains kubevirt v0.51.0, and also testing upgrade from the previous HCO minor version (1.6.z), which contains kubevirt v0.49.0. Both of which are failing on all platforms.

David, is there something special in v0.52.0 that would allow a successful upgrade to v0.53.0 ?

**Comment by xpivarc** [comment]
It seems to me (at least it is the case for https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kubevirt_hyperconverged-cluster-operator/1918/pull-ci-kubevirt-hyperconverged-cluster-operator-main-hco-e2e-upgrade-index-azure/1521232245312983040#1:build-log.txt%3A11105) that you have 3 different cpus(Skylake-Server, Cascadelake-Server, Icelake-Server). This will block the scheduling of the target pod for migration.

**Comment by orenc1** [comment]
This is for this specific run in azure (although we haven't seen this issue before).
The same exact upgrade test also runs on [AWS](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kubevirt_hyperconverged-cluster-operator/1918/pull-ci-kubevirt-hyperconverged-cluster-operator-main-hco-e2e-upgrade-prev-index-aws/1521232245346537472#1:build-log.txt%3A1515) and [GCP](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kubevirt_hyperconverged-cluster-operator/1918/pull-ci-kubevirt-hyperconverged-cluster-operator-main-okd-hco-e2e-upgrade-index-gcp/1521232245493338112#1:build-log.txt%3A1439), and there all of the worker nodes have the same CPU:

AWS:
```
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/kubevirt_hyperconverged-cluster-operator/1918/pull-ci-kubevirt-hyperconverged-cluster-operator-main-hco-e2e-upgrade-prev-index-aws/1521232245346537472/artifacts/hco-e2e-upgrade-prev-index-aws/gather-extra/artifacts/nodes.json | jq .items[].metadata.labels | grep "host-model-cpu.node.kubevirt.io"
  "host-model-cpu.node.kubevirt.io/EPYC": "true",
  "host-model-cpu.node.kubevirt.io/EPYC": "true",
  "host-model-cpu.node.kubevirt.io/EPYC": "true",
```

GCP:
```
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/kubevirt_hyperconverged-cluster-operator/1918/pull-ci-kubevirt-hyperconverged-cluster-operator-main-okd-hco-e2e-upgrade-index-gcp/1521232245493338112/artifacts/hco-e2e-upgrade-index-gcp/gather-extra/artifacts/nodes.json | jq .items[].metadata.labels | grep "host-model-cpu.node.kubevirt.io"
  "host-model-cpu.node.kubevirt.io/Haswell-noTSX-IBRS": "true",
  "host-model-cpu.node.kubevirt.io/Haswell-noTSX-IBRS": "true",
  "host-model-cpu.node.kubevirt.io/Haswell-noTSX-IBRS": "true",
```

And in the virt-controller logs I'm seeing a lot (thousands) of logs of:
```
{"component":"virt-controller","level":"info","msg":"reenqueuing Migration vmsns/kubevirt-workload-update-td8hq","pos":"migration.go:198","reason":"vmi is inelgible for migration because another migration job is running","timestamp":"2022-05-02T22:12:42.011486Z"}
```
[virt-controller logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/kubevirt_hyperconverged-cluster-operator/1918/pull-ci-kubevirt-hyperconverged-cluster-operator-main-okd-hco-e2e-upgrade-index-gcp/1521232245493338112/artifacts/hco-e2e-upgrade-index-gcp/gather-extra/artifacts/pods/kubevirt-hyperconverged_virt-controller-c4d5d6767-6swv7_virt-controller.log)

**Comment by xpivarc** [comment]
@davidvossel Will/did, you have time to have a look?

Otherwise, I would take this later today.

**Comment by xpivarc** [comment]
Fixing in https://github.com/kubevirt/kubevirt/pull/7691

**Comment by orenc1** [comment]
Thanks, @xpivarc .
#7691 has merged. Waiting for a new KV release containing that fix to verify.

**Comment by xpivarc** [comment]
@orenc1 @nunnatsa  Btw. I suspect this was also the root cause why you couldn't bump to 0.52.

**Comment by orenc1** [comment]
we are witnessing the very same issue with the newly released KubeVirt v0.53.0.
I see that #7701 is not included in https://github.com/kubevirt/kubevirt/commits/v0.53.0.

@davidvossel , any plans to release v0.53.1 soon that includes @xpivarc 's fix so we could consume in HCO?

**Comment by xpivarc** [comment]
@orenc1 
The release should be available today https://github.com/kubevirt/kubevirt/releases/tag/v0.53.1

**Comment by xpivarc** [comment]
/close
@orenc1 Release is available https://github.com/kubevirt/kubevirt/releases/tag/v0.53.1
Please reopen if you find any issues.

**Comment by orenc1** [comment]
The reported issue with the upgrade is resolved in v0.53.1, thanks.
However, we can't merge it yet, due to flakiness in some test cases in kubevirt test suite.
I'll open a new issue if needed.

**Comment by xpivarc** [comment]
> The reported issue with the upgrade is resolved in v0.53.1, thanks. However, we can't merge it yet, due to flakiness in some test cases in kubevirt test suite. I'll open a new issue if needed.

Please do and feel free to ping me.
